### DISCUSSION TITLE
PostsService.java#L95-L114

### DISCUSSION THREAD

**Discussion by KangSeongHyo** [dnd-side-project/dnd-5th-3-backend#143] [코드 리뷰]
https://github.com/dnd-side-project/dnd-5th-3-backend/blob/cb1f6c3d27c0978401e01b9eaf19f9a3634c2933/src/main/java/com/dnd5th3/dnd5th3backend/service/PostsService.java#L95-L114

요거는 if문 외에 다양한 방법이 있을 것 같은데 https://techblog.woowahan.com/2527/
참고해보시면 좋을것 같아요

**Comment by zooneon** [comment]
좋은 글 감사합니다 ㅎㅎ
읽어봤는데 위 로직 같은 경우는 db에 직접 접근하는 로직이라 enum으로 어떻게 처리할 수 있을지 감이 오지 않네요...
혹시 좋은 방법 있으시면 공유 부탁드립니다 😭
저는 우선 분기문 제거하고 querydsl을 이용해서 동적으로 가져오도록 수정해 볼 예정입니다!

**Reply by KangSeongHyo** [reply]
처리가 어렵다면 저 String 대신 enum으로 처리하는것도 있을것 같고 interface로 묶어서도 처리가 가능하지 않을까 생각이 드네요
